### PR TITLE
Make namazu package build

### PR DESCRIPTION
**Pull request by jstranik**
Made the package build. Don't use included MMagic module, but instead
use the nix verison which is compatible with the current version of perl.

###### Motivation for this change
The package did not build and was marked broken.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/namazu/default.nix [hunk 1]
```diff
@@ -1,30 +1,21 @@
...
 
   src = fetchurl {
-    url = "http://namazu.org/stable/${name}.tar.gz";
+    url = "http://namazu.org/stable/${pname}-${version}.tar.gz";
     sha256 = "1xvi7hrprdchdpzhg3fvk4yifaakzgydza5c0m50h1yvg6vay62w";
   };
 
-  buildInputs = [ perl ];
+  buildInputs = [ perl makeWrapper ];
+  propagatedBuildInputs = [perlPackages.FileMMagic];
```
Comment (SuperSandro2000) [inline pkgs/tools/text/namazu/default.nix:13]:
```suggestion
  buildInputs = [ perl ];
  nativeBuildInputs = [ makeWrapper ];
```

File: pkgs/tools/text/namazu/default.nix [hunk 2]
```diff
@@ -1,30 +1,21 @@
...
-  # First install the `File::MMagic' Perl module.
-  preConfigure = ''
-    ( cd File-MMagic &&                              \
-      perl Makefile.PL                               \
-        LIB="$out/${perl.libPrefix}/${perl.version}" \
-        INSTALLSITEMAN3DIR="$out/man" &&             \
-      make && make install )
-    export PERL5LIB="$out/${perl.libPrefix}/${perl.version}:$PERL5LIB"
+  postInstall = ''
+      wrapProgram $out/bin/mknmz --set PERL5LIB ${perlPackages.makeFullPerlPath propagatedBuildInputs}
```
Comment (SuperSandro2000) [inline pkgs/tools/text/namazu/default.nix:16]:
```suggestion
      wrapProgram $out/bin/mknmz --set PERL5LIB ${perlPackages.makeFullPerlPath [ perlPackages.FileMMagic ]}
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/text/namazu/default.nix [hunk 3]
```diff
@@ -1,30 +1,21 @@
-{ fetchurl, lib, stdenv, perl }:
+{ fetchurl, lib, stdenv, perl, perlPackages }:
```
Comment (OPNA2608) [inline pkgs/tools/text/namazu/default.nix:1]:
```suggestion
{ fetchurl, lib, stdenv, perl, perlPackages, makeWrapper }:
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

**Comment by jstranik** [comment]
@ofborg eval

File: pkgs/tools/text/namazu/default.nix [hunk 4]
```diff
@@ -1,30 +1,21 @@
...
-  # First install the `File::MMagic' Perl module.
-  preConfigure = ''
-    ( cd File-MMagic &&                              \
-      perl Makefile.PL                               \
-        LIB="$out/${perl.libPrefix}/${perl.version}" \
-        INSTALLSITEMAN3DIR="$out/man" &&             \
-      make && make install )
-    export PERL5LIB="$out/${perl.libPrefix}/${perl.version}:$PERL5LIB"
+  postInstall = ''
+      wrapProgram $out/bin/mknmz --set PERL5LIB ${perlPackages.makeFullPerlPath [ perlPackages.FileMMagic ]}
```
Comment (SuperSandro2000) [inline pkgs/tools/text/namazu/default.nix:16]:
```suggestion
    wrapProgram $out/bin/mknmz --set PERL5LIB ${perlPackages.makeFullPerlPath [ perlPackages.FileMMagic ]}
```

**Pull request merged**